File: Algorithmic Toolbox/Assignment/week3_greedy_algorithms/assignment/Change.java
import java.util.Scanner;

public class Change {
    private static int getChange(int m) {
        //write your code here
        if (m == 0){
            return 0;
        }
        if (m >= 10){
            return getChange(m-10) + 1;
        }
        else if(m >= 5){
            return getChange(m-5) + 1;
        }
        else{
            return getChange(m-1) + 1;
        }

    }
    public void test(){
        System.out.println(getChange(5));
        System.out.println(getChange(154));
        System.out.println(getChange(12));
    }
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int m = scanner.nextInt();
        System.out.println(getChange(m));

    }
}

File: Algorithmic Toolbox/Assignment/week3_greedy_algorithms/1_money_change/change.py
# Uses python3
import sys

def get_change(m):
    #write your code here

    if m == 0:
        return 0
    if m > 10:
        return get_change(m-10) + 1
    elif m > 5:
        return get_change(m-5) + 1
    else:
        return get_change(m-1) + 1


if __name__ == '__main__':
    m = int(sys.stdin.read())
    print(get_change(m))
